flight/Modules/VtolPathFollower/vtol_follower_control.c

```diff
@@ -72,123 +197,184 @@ int32_t vtol_follower_control_path(const float dT, const PathDesiredData *pathDe
 {
 	PositionActualData positionActual;
 	PositionActualGet(&positionActual);
+
+	VelocityActualData velocityActual;
+	VelocityActualGet(&velocityActual);
+
+	const float cur_pos_ned[3] = {
+		positionActual.North +
+		    velocityActual.North*guidanceSettings.PositionFeedforward,
+		positionActual.East +
+		    velocityActual.East*guidanceSettings.PositionFeedforward,
+		positionActual.Down };
+
+	path_progress(pathDesired, cur_pos_ned, progress);
 	
-	const float cur[3] = { positionActual.North, positionActual.East, positionActual.Down };
-	
-	path_progress(pathDesired, cur, progress);
-	
+	// Interpolate desired velocity and altitude along the path
+	float groundspeed = vtol_interpolate(progress->fractional_progress,
+	    pathDesired->StartingVelocity, pathDesired->EndingVelocity);
+
+	if (progress->fractional_progress >= 1)
+		groundspeed = 0;
```

Comment:
I don’t like using this setting groundspeed to 0 as a hook to trigger ln 246, especially given this is essentially a known bug to do this. can you just use ```progress->fractional_progress >= 1)``` there? better yet since we have logic checks on whether fraction_progress is > / < 1 on line 217, 234, 242, 246 (implicit), could you just consolidate them into two if blocks? there is no need to compute these things when we simply ```return vtol_follower_control_simple(dT, pathDesired->End, 0, 1);```
see 16e2a8455e4b466cee7669e768ac9416d92c757ag in the PR against this branch on your repo